File: robot-factory/typescript/src/PartQuote.ts
import type { Money } from './Money.js';
import type { PartOption } from './PartOption.js';
import type { PartType } from './PartType.js';

export interface PartQuote {
  readonly type: PartType;
  readonly option: PartOption;
  readonly price: Money;
  readonly supplierName: string;
}

File: robot-factory/python/src/robot_factory/part_quote.py
from __future__ import annotations

from dataclasses import dataclass

from .money import Money
from .part_option import PartOption
from .part_type import PartType


@dataclass(frozen=True)
class PartQuote:
    type: PartType
    option: PartOption
    price: Money
    supplier_name: str
